src/window/mod.rs

```diff
@@ -253,7 +324,18 @@ impl<F: Frame + 'static> Window<F> {
     /// You need to call `refresh()` afterwards for this to properly
     /// take effect.
     pub fn set_decorate(&self, decorate: bool) {
+        use self::zxdg_toplevel_decoration_v1::Mode;
         self.frame.lock().unwrap().set_hidden(!decorate);
+        if let Some(ref dec) = self.decoration {
+            if decorate {
+                // let the server decide decorations
+                dec.unset_mode();
+            } else {
+                // we must be the decorators, to be able to not
+                // show any decoration
+                dec.set_mode(Mode::ClientSide);
```

Comment:
It might be more complicated to implement, but if you don't want to be decorated, you should destroy the decoration object. The compositor always has the last word wrt decoration mode.

Reply:
Right, indeed. This will be more convoluted to implement though.